[fix-10729]The request's url given by the test case looks different f…

fuchanghai opened the pull request:
…rom the controller layer. Some usages and parameter format errors




## Purpose of the pull request
fix #10729  only in class DataSourceControllerTest


## Brief change log

The request's url given by the test case looks different from the controller layer. Some usages and parameter format errors in class DataSourceControllerTest
## Verify this pull request





This pull request is already covered by existing tests, which annotated by  org.junit.Ignore  in class DataSourceControllerTest

Commit:
[fix-10729]The request's url given by the test case looks different from the controller layer. Some usages and parameter format errors

SbloodyS commented:
> Kudos, SonarCloud Quality Gate passed!    [![Quality Gate passed](https://camo.githubusercontent.com/e4a90cf346b46d44c5652111d68603714bfb7693043d07a95992de3a30bfce04/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f7061737365642d313670782e706e67)](https://sonarcloud.io/dashboard?id=apache-dolphinscheduler&pullRequest=10734)
> 
> [![Bug](https://camo.githubusercontent.com/4c6102327f5a954f9c8acaf2e2714183157a9e41717b371b2cd585cf25057310/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6275672d313670782e706e67)](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=BUG) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=BUG) [![Vulnerability](https://camo.githubusercontent.com/3ba1ee49636ffc3427e38649a9f8a65ee392f28e8a662fcf96ce24cefbb520e9/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f76756c6e65726162696c6974792d313670782e706e67)](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=VULNERABILITY) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=VULNERABILITY) [![Security Hotspot](https://camo.githubusercontent.com/fb735cbe76f8d5e1679c76ce83b740ceb1eaf62de4f7bf88623dc9953261aff7/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f73656375726974795f686f7473706f742d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=SECURITY_HOTSPOT) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=SECURITY_HOTSPOT) [![Code Smell](https://camo.githubusercontent.com/8fe18b2dfb6f7d4e44582f281b29f617eb5ae07c248d2002ca586e91da219212/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f636f64655f736d656c6c2d313670782e706e67)](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=CODE_SMELL) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=CODE_SMELL)
> 
> [![No Coverage information](https://camo.githubusercontent.com/86785eaa572e061f5ecb3f8bf1cbb7b8eca44925083f92d067805b4362cead0c/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f436f76657261676543686172742f4e6f436f766572616765496e666f2d313670782e706e67)](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10734&metric=coverage&view=list) No Coverage information [![0.0%](https://camo.githubusercontent.com/8047c63e1f9ed03f63001e1eadce4676bade3e0f83ec690a9c625287796248a6/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f4475706c69636174696f6e732f332d313670782e706e67)](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10734&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10734&metric=new_duplicated_lines_density&view=list)

Please pay attention to code smell.

Commit:
[fix-10729]The request's url given by the test case looks different from the controller layer. Some usages and parameter format errors

Commit:
[fix-10729]The request's url given by the test case looks different from the controller layer. Some usages and parameter format errors

Commit:
[fix-10729]The request's url given by the test case looks different from the controller layer. Some usages and parameter format errors

Commit:
[fix-10729]fix the code smell

fuchanghai commented:
Now there is no code smell in my local area, and it has been pushed up. It seems that there are still three excepted. Is there anything else I should pay attention to? @SbloodyS

SbloodyS commented:
> Now there is no code smell in my local area, and it has been pushed up. It seems that there are still three excepted. Is there anything else I should pay attention to? @SbloodyS

Since you are a first-time-contributor. You do not have permission to run CI. You can ping me when you want to run CI.

fuchanghai commented:
## When I decorate a class or method with public, or have a code smell
![image](https://user-images.githubusercontent.com/33984497/177238866-450c7625-b06d-4c4f-9f9c-a3d1cad18be0.png)

## when i remove public,CI will error
@SbloodyS

SbloodyS commented:
> ## When I decorate a class or method with public, or have a code smell
> ![image](https://user-images.githubusercontent.com/33984497/177238866-450c7625-b06d-4c4f-9f9c-a3d1cad18be0.png)
> 
> ## when i remove public,CI will error
> @SbloodyS

The reason is that you used the public method in class.

Commit:
[fix-10729]decorate DataSourceControllerTest with public

fuchanghai commented:
Excuse me, please help me to run  CI

fuchanghai commented:
@SbloodyS

SbloodyS commented:
> Excuse me, please help me to run CI

I had run CI.

SbloodyS commented on dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/DataSourceControllerTest.java:
```diff
@@ -19,19 +19,23 @@
-import org.junit.Assert;
-import org.junit.Ignore;
-import org.junit.Test;
+import org.junit.*;
```
Please avoid import *. This violates checkstyle.

Commit:
[fix-10729]replace import *

fuchanghai commented:
@SbloodyS i had remove import *.thanks for your patience and help.please run CI again

SbloodyS commented:
> @SbloodyS i had remove import *.thanks for your patience and help.please run CI again

I had approved to run CI.

fuchanghai commented:
@SbloodyS If you have time, can you help me take a look. This error does not seem to be a code error, but the test case does not exist in the database, or the database information cannot be connected

SbloodyS commented:
> @SbloodyS If you have time, can you help me take a look. This error does not seem to be a code error, but the test case does not exist in the database, or the database information cannot be connected

You are missing the logic of MySQL database initialization. It is recommended that you debug and execute unit tests locally.

Commit:
[fix-10729]

fuchanghai commented:
> > @SbloodyS If you have time, can you help me take a look. This error does not seem to be a code error, but the test case does not exist in the database, or the database information cannot be connected
> 
> You are missing the logic of MySQL database initialization. It is recommended that you debug and execute unit tests locally.

i'm so sorry for that. sonar recommended that remove 'public' modifier from method and class. Otherwise, it will prompt code smells.After I removed it, I didn't test it, which resulted in an error when I removed it.(initialzationError  which beacuse of remove 'public' modifier ). i had tested in my local. 
there only two mistake that not seem to be a code error ( can't find resource by datasouceId  and can't connect database) please run CI

fuchanghai commented:
![image](https://user-images.githubusercontent.com/33984497/177974756-631c0a86-6194-433b-80bc-6247a58a7cbf.png)
it really work in my local 。there mistake which  is can't find resouce by id. should i add @Ignore?

fuchanghai commented:
![image](https://user-images.githubusercontent.com/33984497/177976690-64fcfa7b-249c-4bd2-9eba-bd6f98d79748.png)
@ParameterizedTest 不能和@Test 一起使用。if idon't use @ParameterizedTest on some method. will have code smells.i don't know how to balance them。can you give me some adivce?

Commit:
[fix-10729]

SbloodyS commented:
It seems the error the reason for the error is that the checkstyle check failed. You can use `https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html` to check it locally.

Commit:
[fix-10729]

Commit:
[fix-10729]checkstyle

fuchanghai commented:
@SbloodyS

fuchanghai commented:
@SbloodyS i think i should not replace these 3 tests with a single Parameterized one.because here are three interfaces with different functions

SbloodyS commented:
> @SbloodyS i think i should not replace these 3 tests with a single Parameterized one.because here are three interfaces with different functions

Okey. That's makes sense to me.

SbloodyS reviewed (approved):
LGTM

fuchanghai commented:
@caishunfeng please review

caishunfeng commented:
Thanks for your contribution @fuchanghai :smile:

EricGao888 commented on dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/DataSourceControllerTest.java:
```diff
@@ -17,21 +17,26 @@
 import org.junit.Assert;
 import org.junit.Ignore;
 import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
```
@fuchanghai We are not supposed to have both jUnit5 and jUnit4 style UTs in one file.
